here-naksha-storage-http/src/main/java/com/here/naksha/storage/http/POpToQuery.java

```diff
@@ -0,0 +1,178 @@
+
+  public static final String OR_DELIMITER = ",";
+  public static final String PATH_SEGMENT_DELIMITER = ".";
+
+  static String p0pToQuery(POp pOp) {
+    if (pOp.op() == AND) return and(pOp);
+    else return pOpToMultiValueComparison(pOp).resolve();
+  }
+
+  private static String and(POp pOp) {
+    assertHasAtLeastOneChildren(pOp);
+    return pOp.children().stream().map(POpToQuery::p0pToQuery).collect(Collectors.joining(AND_DELIMITER));
+  }
+
+  //
+  // MultiValueComparison operations
+  //
+  private static MultiValueComparison pOpToMultiValueComparison(POp pOp) {
+    if (pOp.op() == AND) throw unsupportedOperation("AND can be only a top level operation");
+    if (pOp.op() == OR) return or(pOp);
+    if (pOp.op() == NOT) return not(pOp);
+    if (pOp.op() == EXISTS) return exists(pOp);
+    return simpleLeafOperator(pOp);
+  }
+
+  private static MultiValueComparison or(POp pOp) {
+    assertHasAtLeastOneChildren(pOp);
+
+    return pOp.children().stream()
+        .map(POpToQuery::pOpToMultiValueComparison)
+        .reduce((l, r) -> {
+          if (!Objects.equals(l.operator, r.operator))
+            throw unsupportedOperation(
+                "Operators " + l.operator + " and " + r.operator + " combined in one OR");
+          if (!Objects.equals(l.path, r.path)) throw unsupportedOperation("Paths in OR are not equal");
+          return new MultiValueComparison(l.operator, l.path, ArrayUtils.addAll(l.values, r.values));
+        })
+        .get();
+  }
+
+  private static MultiValueComparison not(POp pOp) {
+    assertHasNChildren(pOp, 1);
+    MultiValueComparison multiValueComparison =
+        pOpToMultiValueComparison(pOp.children().get(0));
+    String newOperator =
+        switch (multiValueComparison.operator) {
+          case EQ_OPERATOR -> NOT_EQ_OPERATOR;
+          case NOT_EQ_OPERATOR -> EQ_OPERATOR;
+          default -> throw unsupportedOperation(multiValueComparison.operator);
+        };
+    return new MultiValueComparison(newOperator, multiValueComparison.path, multiValueComparison.values);
+  }
+
+  //
+  // Leaf operators
+  //
+  private static MultiValueComparison exists(POp pOp) {
+    assertHasNChildren(pOp, 0);
+    assertHasPathSet(pOp);
+
+    return new MultiValueComparison(NOT_EQ_OPERATOR, pOp.getPropertyRef().getPath(), NULL);
+  }
+
+  private static MultiValueComparison simpleLeafOperator(POp pOp) {
+    assertHasNChildren(pOp, 0);
+    assertHasPathSet(pOp);
+    assertHasValueSet(pOp);
+
+    String operator = simpleLeafOperators.get(pOp.op());
+    if (operator == null) throw unsupportedOperation(pOp.op() + " not supported");
+    return new MultiValueComparison(
+        operator, pOp.getPropertyRef().getPath(), pOp.getValue().toString());
+  }
+
+  private static final Map<POpType, String> simpleLeafOperators = Map.of(
+      EQ, EQ_OPERATOR,
+      GT, "=gt=",
+      GTE, "=gte=",
+      LT, "=lt=",
+      LTE, "=lte=",
+      CONTAINS, "=cs=");
```

Comment:
As this is a constant field consider
- moving it to top
- applying upper case (`SIMPLE_LEAF_OPERATORS`)

for reference see file organization rules that IMO make much sense: 
https://www.oracle.com/java/technologies/javase/codeconventions-fileorganization.html

Reply:
Done